diff --git a/input/drivers/android_input.c b/input/drivers/android_input.c
index e555e11b8e..8e9a442566 100644
--- a/input/drivers/android_input.c
+++ b/input/drivers/android_input.c
@@ -1588,6 +1588,13 @@ static INLINE void android_input_poll_event_type_key(
       AInputEvent *event, int port, int keycode, int source,
       int type_event, int *handled)
 {
+#ifdef HAVE_RICOTTA_IGM
+   if (ricotta_bridge_intercept_key(keycode, AKeyEvent_getAction(event)))
+   {
+      *handled = 1;
+      return;
+   }
+#endif
    uint8_t *buf;
    int action           = AKeyEvent_getAction(event);
    int keysym           = keycode;
