diff --git a/input/drivers/android_input.c b/input/drivers/android_input.c
index df37d1a051..2f2db2c6f2 100644
--- a/input/drivers/android_input.c
+++ b/input/drivers/android_input.c
@@ -1024,6 +1024,13 @@ static INLINE void android_input_poll_event_type_key(
       AInputEvent *event, int port, int keycode, int source,
       int type_event, int *handled)
 {
+#ifdef HAVE_RICOTTA_IGM
+   if (ricotta_bridge_intercept_key(keycode, AKeyEvent_getAction(event)))
+   {
+      *handled = 1;
+      return;
+   }
+#endif
    uint8_t *buf = android_key_state[port];
    int action   = AKeyEvent_getAction(event);
    int keysym   = keycode;
@@ -1827,6 +1834,9 @@ static void android_input_reinit(void)
  */
 static void android_input_poll(void *data)
 {
+#ifdef HAVE_RICOTTA_IGM
+   ricotta_bridge_poll_commands();
+#endif
    int ident;
    struct android_app *android_app = (struct android_app*)g_android;
    android_input_t *android        = (android_input_t*)data;
